src/widgets/paragraph.rs

```diff
@@ -203,12 +201,476 @@ impl<'a> Widget for Paragraph<'a> {
 #[cfg(test)]
 mod test {
     use super::*;
+    use crate::backend::TestBackend;
+    use crate::{
+        style::Color,
+        text::{Span, Spans},
+        widgets::Borders,
+        Terminal,
+    };
+
+    fn test_case(paragraph: &Paragraph, expected: Buffer) {
+        let backend = TestBackend::new(expected.area.width, expected.area.height);
+        let mut terminal = Terminal::new(backend).unwrap();
+
+        terminal
+            .draw(|f| {
+                let size = f.size();
+                f.render_widget(paragraph.clone(), size);
+            })
+            .unwrap();
+
+        terminal.backend().assert_buffer(&expected);
```

Comment:
For sole testing purposes a direct buffer would suffice. I used the `TestBackend` because it formats the error in case of a failure more nicely. I think your #161 may to the same if I understood correctly? I will look into its usage for these tests.

Reply:
For the majority of these tests the extra information given by the `assert_buffer_eq!` macro is irrelevant, as we do not make use styles apart from one test method. I will leave this for now, since the current error is more concise and easier to read, but it would be of great use in a crate public module containing common testing methods.

Reply:
I was thinking about this a little more. Combining the way you wrote these tests with a macro specific to testing widget rendering could be the best of both worlds. e.g. something like:
```
assert_widget_renders!(
    paragraph.new("line 1\nline 2"),
    ["line 1  ","line 2  "],
    [(0, 1, 6, 1, Style::default().bg(Color::Blue))],
    "some description");
```

Reply:
That seems like a great addition! Unfortunately I have absolutely no experience in writing macros... 😂